main/main.js

```diff
@@ -0,0 +1,171 @@
+const searchButton = document.getElementById("search-button");
+const input = document.getElementById("city-name");
+const weather = document.getElementById("weather");
+const temperature = document.getElementById("temperature");
+const weatherDescription = document.getElementById("weather-description");
+const city = document.getElementById("city");
+const country = document.getElementById("country");
+const iconImage = document.getElementById("icon-image");
+const errorMessage = document.getElementById("error-message");
+const weatherContainer = document.getElementById("weather-container");
+const windSpeed = document.getElementById("wind-speed");
+const humidity = document.getElementById("humidity");
+const card = document.getElementById("card");
+const spinner = document.getElementById("spinner");
+
+const key = `82005d27a116c2880c8f0fcb866998a0`;
+
+const weatherTypes = [
+  "Clear",
+  "Clouds",
+  "Rain",
+  "Snow",
+  "Thunderstorm",
+  "Drizzle",
+  //7xx: Atmosphere there are many like Smoke, Mist, Haze, Fog etc will come in same category
+];
+
+const fetchData = async (lat, lon) => {
+  try {
+    const response = await fetch(
+      `http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${key}`
+    );
+    if (response.status === 200) {
+      const data = await response.json();
+      const description = data.weather[0].description;
+      const tempInKelvin = data.main.temp;
+      const tempInCelcius = Math.floor(tempInKelvin - 273.15);
+      const cityName = data.name;
+      const countryName = data.sys.country;
+      const weatherType = data.weather[0].main;
+      const iconId = data.weather[0].icon;
+      const humid = data.main.humidity;
+      const wind = data.wind.speed;
+
+      return {
+        description,
+        tempInCelcius,
+        cityName,
+        countryName,
+        weatherType,
+        iconId,
+        humid,
+        wind,
+      };
+    } else {
+      console.log("ERROR! Wrong City Name");
+    }
+  } catch (e) {
+    console.log(e);
```

Comment:
yu should also return something in case of error in Promise type functions @aayushbatra02 
or throw error